tests/test_ecephys/test_tools_spikeinterface.py

```diff
@@ -837,6 +837,41 @@ def test_row_matching_by_channel_name_with_new_property(self):
         self.assertListEqual(list(self.nwbfile.electrodes["channel_name"].data), expected_names)
         self.assertListEqual(list(self.nwbfile.electrodes["property"].data), expected_property_values)
 
+    def test_property_metadata_mismatch(self):
+        """
+        Adding recordings that do not contain all properties described in
+        metadata should not error.
+        """
+        self.recording_1.set_property(key="common_property", values=["value_1"] * self.num_channels)
+        self.recording_2.set_property(key="common_property", values=["value_2"] * self.num_channels)
+        self.recording_1.set_property(key="property_1", values=[f"value_{n+1}" for n in range(self.num_channels)])
+        self.recording_2.set_property(key="property_2", values=[f"value_{n+1}" for n in range(self.num_channels)])
+
+        metadata = dict(
+            Ecephys=dict(
+                Electrodes=[
+                    dict(name="common_property", description="no description."),
+                    dict(name="property_1", description="no description."),
+                    dict(name="property_2", description="no description."),
+                ]
+            )
+        )
+
+        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile, metadata=metadata)
+        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile, metadata=metadata)
+
+        actual_common_property_values = list(self.nwbfile.electrodes["common_property"].data)
+        expected_common_property_values = ["value_1", "value_1", "value_1", "value_1", "value_2", "value_2"]
```

Comment:
Not important in this PR but I wanted to highlight this maybe for quick discussion or to open an issue. @CodyCBakerPhD.

The behavior of add_electrodes is non-conmutative here with respect to the adddition of properties. That is, if you have added the electrodes in the different order the result would have been different.

``` python
        add_electrodes(recording=self.recording_2, nwbfile=self.nwbfile, metadata=metadata)
        add_electrodes(recording=self.recording_1, nwbfile=self.nwbfile, metadata=metadata)

        actual_common_property_values = list(self.nwbfile.electrodes["common_property"].data)
        expected_common_property_values = ["value_1", "value_1", "value_2", "value_2", "value_2", "value_2"]
       # Instead of ["value_1", "value_1", "value_1", "value_1", "value_2", "value_2"] as it is now
```
Am I correct, @felixp8?
Is this some behavior that we want? Do you guys think that this grants a warning for the user? something else?